### PR TITLE
Fix issue #404 - Allow to use bazel subtarget

### LINKED ISSUE #404: Non-top-level bazel builds don't properly tag artifact
### Expected behavior

When a bazel artifact has a non-top-level target, e.g. `//botbot:botbot.tar`, and an `imageName` like `gcr.io/my-project/botbot`, then running `skaffold dev` should produce a docker image named `gcr.io/my-project/botbot` tagged with the appropriate digest.

### Actual behavior

Running `skaffold dev` generates a warning (`WARN[0009] run: build: build step: digest not found`) and produces a docker image named `bazel/botbot` tagged with `botbot`.

### Information

```
Skaffold version: v0.4.0
Operating system: Mac OS
Content of skaffold.yaml:

apiVersion: skaffold/v1alpha2
kind: Config
build:
  artifacts:
  - imageName: gcr.io/my-project/botbot
    workspace: .
    bazel:
      target: botbot/botbot.tar
  local: {}
deploy:
  helm:
    releases:
    - name: botbot
      chartPath: botbot/helm
      namespace: botbot
      values:
        image: gcr.io/my-project/botbot
```

### Steps to reproduce the behavior

My project consists of the following files:

`WORKSPACE`
```
workspace(
    name="botbot",
)

git_repository(
    name = "io_bazel_rules_docker",
    remote = "https://github.com/bazelbuild/rules_docker.git",
    tag = "v0.4.0",
)

load(
    "@io_bazel_rules_docker//container:container.bzl",
    "container_image",
    "container_pull",
    container_repositories = "repositories",
)

container_repositories()


container_pull(
    name = "debian",
    registry = "index.docker.io",
    repository = "library/debian",
    tag = "stretch-20180312",
)
```

`botbot/BUILD`
```
package(default_visibility = ["//visibility:public"])

load(
    "@io_bazel_rules_docker//container:container.bzl",
    "container_image",
)

container_image(
    name = "botbot",
    base = "@debian//image",
    files = ["README.md"],
)
```

`botbot/README.md`
```
# TEST
```

And the `skaffold.yaml` file specified above.

Running `skaffold dev` produces the following results:

```
$ skaffold dev
Starting build...
Found [minikube] context, using local docker daemon.
INFO: Analysed target //botbot:botbot.tar (2 packages loaded).
INFO: Found 1 target...
Target //botbot:botbot.tar up-to-date:
  bazel-bin/botbot/botbot.tar
INFO: Elapsed time: 7.307s, Critical Path: 5.40s
INFO: Build completed successfully, 8 total actions
{"stream":"Loaded image: bazel/botbot:botbot\n"}
Digest: bazel:botbot/botbot
WARN[0009] run: build: build step: digest not found
Watching for changes...
```

And running `docker images` produces:

```
$ docker images
REPOSITORY                                 TAG                                                                IMAGE ID            CREATED             SIZE
bazel/botbot                               botbot                                                             21aa76377d9f        48 years ago        100MB
```

### PR DESCRIPTION
**Pull request by fclaeys**
@dgageot @r2d4 Here is a first implementation to fix #404 
This is my first contribution on the project do not hesitate to make me your remarks to improve the code.

### CODE REVIEW

File: pkg/skaffold/build/bazel.go [hunk 1]
```diff
@@ -30,6 +30,37 @@ import (
 	"github.com/pkg/errors"
 )
 
+func trimTarget(buildTarget string) string {
+
+	//TODO(r2d4): strip off leading //:, bad
+	trimedTarget := strings.TrimPrefix(buildTarget, "//")
```
Comment (dlorenc) [inline pkg/skaffold/build/bazel.go]: nit: typos. Could you rename these to "trimmed"?

Comment (fclaeys) [reply pkg/skaffold/build/bazel.go]: Done :)

**Comment by dgageot** [comment]
@fclaeys Can you please rebase to both update the branch and run the windows tests? Thanks!

**Comment by fclaeys** [comment]
@dgageot branch updated

**Pull request merged**